### DISCUSSION TITLE
device-sdk-c  Interface documentation

### DISCUSSION THREAD

**Discussion by wangqingpen** [edgexfoundry/community#261] [Q&A]
https://github.com/edgexfoundry/device-sdk-c/blob/main/src/c/examples/template.c   
This file has documentation of whether or not to call the interface, such as (devsdk_protocols *) device->address

**Comment by jiekechoo** [comment]
@wangqingpen What kind of interface do you need?

BTW, if you need to ask some questions, you have to explain the details. According to your previous questions in the discusstions list, most of them were confused to the community.

There are a few tips for you:
- Read more documentation in the docs website: https://docs.edgexfoundry.org and the wiki website: https://wiki.edgexfoundry.org
- Practics more about the sample code, drill-down to the source code for the device sdk
- Try to write your custom device service based on the sample repository
- Learn more IoT basic knowledge especially the edge computing
